resharper/resharper-unity/src/Yaml/Psi/Caches/MetaFileGuidCache.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using JetBrains.DataFlow;
+using JetBrains.ReSharper.Plugins.Yaml.Psi;
+using JetBrains.ReSharper.Plugins.Yaml.Psi.Tree;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.Caches;
+using JetBrains.ReSharper.Psi.Files;
+using JetBrains.Util;
+using JetBrains.Util.DataStructures;
+
+namespace JetBrains.ReSharper.Plugins.Unity.Yaml.Psi.Caches
+{
+    [PsiComponent]
+    public class MetaFileGuidCache : SimpleICache<MetaFileCacheItem>
+    {
+        // We expect to only get one asset with a given guid, but copy/pasting .meta files could break that.
+        // CompactOneToListMap is optimised for the typical use case of only one item per key
+        private readonly CompactOneToListMap<string, FileSystemPath> myAssetGuidToAssetFilePaths =
+            new CompactOneToListMap<string, FileSystemPath>();
+
+        public MetaFileGuidCache(Lifetime lifetime, IPersistentIndexManager persistentIndexManager)
+            : base(lifetime, persistentIndexManager, MetaFileCacheItem.Marshaller)
+        {
+#if DEBUG
```

Comment:
Because it clears the cache on each load. Most useful for when you modify the contents of the cache without updating the version number (i.e. so you don't increment it 20 times before you commit :smile:)